src/lib.rs

```diff
@@ -110,7 +110,45 @@ pub trait Curve:
     }
 
     /// Converts this element into its affine representation.
-    fn to_affine(&self) -> Self::AffineRepr;
+    fn to_affine(&self) -> Self::Affine;
+}
+
+/// Affine representation of an elliptic curve point.
+pub trait CurveAffine:
+    GroupEncoding
+    + Copy
+    + Clone
+    + Sized
+    + Send
+    + Sync
+    + fmt::Debug
+    + PartialEq
+    + Eq
+    + 'static
+    + Neg<Output = Self>
+    + Mul<<Self::Curve as Group>::Scalar, Output = Self::Curve>
+    + for<'r> Mul<&'r <Self::Curve as Group>::Scalar, Output = Self::Curve>
+{
+    /// The efficient representation for this elliptic curve.
+    type Curve: Curve<Affine = Self, Scalar = Self::Scalar>;
+
+    /// Scalars modulo the order of this group's scalar field.
+    ///
+    /// This associated type is temporary, and will be removed once downstream users have
+    /// migrated to using `Curve` as the primary generic bound.
+    type Scalar: PrimeField;
```

Comment:
I want to remove this associated type (and instead use `<Self::Curve as Group>::Scalar`, but for annoying historical type-checking reasons we used (the equivalent of) `C: PrimeCurveAffine` as the main generic parameter in https://github.com/zcash/halo2. I want to change it to use `C: PrimeCurve` instead (or `G: PrimeGroup` where I can), but before I can do that I need to kill the `CurveExt` trait (https://github.com/zcash/pasta_curves/issues/41). So this associated type will stay until that is complete (though I hope to complete 41 in the same release cycle as this PR).